.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,42 @@
+name: "Pull Request Build"
+
+on:
+  pull_request:
+    branches:
+      - master
+      - main
+      # For Testing!
+      - wip-test-containers
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - id: setup-java-11
+        name: Setup Java 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - uses: burrunan/gradle-cache-action@v1.6
+        with:
+          remote-build-cache-proxy-enabled: false
+          arguments: test
+  style:
+    name: Stylecheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - id: setup-java-11
+        name: Setup Java 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - uses: burrunan/gradle-cache-action@v1.6
+        with:
+          remote-build-cache-proxy-enabled: false
+          arguments: spotlessCheck
```

Comment:
At the risk of bikeshedding, do we want newlines at the end of files? 😃

Reply:
You can configure that in spotless.   Separate PR for that, and I'll give you my tirade on style later :)